frameworks/Rust/actix/src/db_pg_direct.rs

```diff
@@ -0,0 +1,154 @@
+use std::io;
+
+use actix::prelude::*;
+use futures::{stream, Future, Stream};
+use rand::{thread_rng, Rng};
+use tokio_postgres::{connect, Client, Statement, TlsMode};
+
+use models::{Fortune, World};
+
+/// Postgres interface
+pub struct PgConnection {
+    cl: Client,
+    fortune: Statement,
+    world: Statement,
+    update: Statement,
+}
+
+impl PgConnection {
+    pub fn connect(db_url: &str) -> impl Future<Item=PgConnection, Error=()> {
+        let hs = connect(db_url.parse().unwrap(), TlsMode::None);
+
+        hs.map_err(|_| panic!("can not connect to postgresql"))
+            .and_then(|(cl, conn)| {
+                Arbiter::spawn(conn.map_err(|e| panic!("{}", e)));
+                cl.prepare("SELECT id, message FROM fortune")
+                    .map_err(|_| ())
+                    .and_then(move |fortune| {
+                        cl.prepare("SELECT id, randomnumber FROM world WHERE id=$1")
+                            .map_err(|_| ())
+                            .and_then(move |world| {
+                                let st = (fortune, world);
+                                cl.prepare("SELECT id FROM world WHERE id=$1")
```

Comment:
Out of curiosity, what is this query used for?

Reply:
it is for db updates, line 96